server/api/scheduler.go

```diff
@@ -0,0 +1,88 @@
+// Copyright 2016 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package api
+
+import (
+	"net/http"
+
+	"github.com/gorilla/mux"
+	"github.com/pingcap/pd/server"
+	"github.com/unrolled/render"
+)
+
+type schedulerHandler struct {
+	*server.Handler
+	r *render.Render
+}
+
+func newSchedulerHandler(handler *server.Handler, r *render.Render) *schedulerHandler {
+	return &schedulerHandler{
+		Handler: handler,
+		r:       r,
+	}
+}
+
+func (h *schedulerHandler) List(w http.ResponseWriter, r *http.Request) {
+	schedulers, err := h.GetSchedulers()
+	if err != nil {
+		h.r.JSON(w, http.StatusInternalServerError, err.Error())
+		return
+	}
+	h.r.JSON(w, http.StatusOK, schedulers)
+}
+
+func (h *schedulerHandler) Post(w http.ResponseWriter, r *http.Request) {
+	var input map[string]interface{}
+	if err := readJSON(r.Body, &input); err != nil {
+		h.r.JSON(w, http.StatusInternalServerError, err.Error())
+		return
+	}
+
+	name, ok := input["name"].(string)
+	if !ok {
+		h.r.JSON(w, http.StatusBadRequest, "missing scheduler name")
+		return
+	}
+
+	switch name {
+	case "leader-balancer":
+		if err := h.AddLeaderBalancer(); err != nil {
```

Comment:
Now we can use the API to delete the leader-balancer and add it again.
For example, if I add a random-leader-scheduler, it will conflict with the leader-balancer, so I can remove the leader-balancer. If I add a shuffle-leader-scheduler, it can run with the leader-balancer concurrently.